src/dom-manipulation.ts

```diff
@@ -0,0 +1,293 @@
+import { combineLatestWith } from "rxjs";
+import { IFlight } from "./models/flight";
+import { fetchStream$, selectedFlightIcoa24$} from "./services/flight-service";
+import {
+  resetMapLocationView,
+  setMapAndMarkerToCurrentFlightLocation,
+} from "./services/leaflet-service";
+import {
+  calculateDirection,
+  convertMeterPerSecondToKilomentersPerHour,
+} from "./utils/utils";
+
+
+class FlightInfoTable extends HTMLElement {
+  constructor() {
+  super();
+  // Create the Shadow Root
+  const shadowRoot = this.attachShadow({ mode: "open" });
+  shadowRoot.innerHTML += `
+  <style>
+    .single-flight {
+      display: grid;
+      grid-template-columns: repeat(4, minmax(0, 1fr));
+      padding-top: 0.25rem;
+      padding-bottom: 0.25rem;
+      font-size: 0.8rem;
+      padding-left: 0.2rem;
+    }
+
+    .vertical-rate {
+      display: none;
+    }
+
+    .direction {
+      display: none;
+    }
+
+    .altitude {
+      display: none;
+    }
+
+    .track-button {
+      text-transform: uppercase;
+      font-weight: 700;
+      cursor: pointer;
+      border-style: none;
+      background: transparent;
+      font-size: 0.8rem;
+      color: #faf755;
+    }
+
+    @media (min-width: 480px) {
+      .single-flight {
+        grid-template-columns: repeat(5, minmax(0, 1fr));
+      }
+
+      .direction {
+        display: contents;
+      }
+    }
+
+    @media (min-width: 769px) {
+      .single-flight {
+        grid-template-columns: repeat(6, minmax(0, 1fr));
+      }
+
+      .altitude {
+        display: contents;
+      }
+    }
+
+    @media (min-width: 1024px) {
+      .single-flight {
+        grid-template-columns: repeat(7, minmax(0, 1fr));
+      }
+
+      .vertical-rate {
+        display: contents;
+      }
+    }
+  </style>`;
+  }
+}
+// Register the custom element
+customElements.define("flight-info-table", FlightInfoTable);
+
+// Sybscribe to api output and update table with new or updated info
+export const flightArraySubcription = fetchStream$.pipe(combineLatestWith(selectedFlightIcoa24$))
+.subscribe(([flightArray, selectedFlight]: [IFlight[], string]) => {
+  const shadowHost = document.querySelector("#flights-info");
```

Comment:
did you clean this up?

Reply:
It is constantly polling every 30 sec, I spoke to Tim he said the only place would be on unload.